openmc/kinetics/solver.py

```diff
@@ -0,0 +1,886 @@
+        f.attrs['chi_delayed_by_mesh'] = self.chi_delayed_by_mesh
+        f.attrs['use_agd'] = self.use_agd
+        f.attrs['use_pcmfd'] = self.use_pcmfd
+        f.attrs['num_delayed_groups'] = self.num_delayed_groups
+        f.attrs['num_outer_time_steps'] = self.num_outer_time_steps
+        f.attrs['use_pregenerated_sps'] = self.use_pregenerated_sps
+        f.attrs['constant_seed'] = self.constant_seed
+        f.attrs['seed'] = self.seed
+        f.attrs['core_volume'] = self.core_volume
+        f.attrs['k_crit'] = self.k_crit
+        f.attrs['method'] = self.method
+        f.attrs['pnl'] = self.states['START'].pnl
+        f.require_group('clock')
+        f['clock'].attrs['dt_outer'] = self.clock.dt_outer
+        f['clock'].attrs['dt_inner'] = self.clock.dt_inner
+        f.require_group('OUTER_STEPS')
+        f.require_group('INNER_STEPS')
+        f.create_dataset('beta', data=self.states['START'].beta())
+        f.create_dataset('beta_shape', data=self.states['START'].beta_shape())
+        f.create_dataset('beta_eff', data=self.states['START'].beta_eff)
+        f.create_dataset('power', data=self.states['START'].power)
+        f.create_dataset('decay_rate', data=self.states['START'].decay_rate)
+        f.create_dataset('inverse_velocity', data=self.states['START'].inverse_velocity)
+        total = self.states['START'].absorption + self.states['START'].outscatter
+        f.create_dataset('total', data=total)
+
+        f.close()
+
+    def run_openmc(self, time_point, method='ADIABATIC'):
+
+        self.setup_openmc(time_point, method)
+
+        # Names of the statepoint and summary files
+        sp_old_name = '{}/statepoint.{}.h5'.format(self.directory, self.settings_file.batches)
+        sp_new_name = '{}/statepoint_{:.6f}_sec.{}.h5'\
+                      .format(self.directory, self.clock.times[time_point], self.settings_file.batches)
+        sum_old_name = '{}/summary.h5'.format(self.directory)
+        sum_new_name = '{}/summary_{:.6f}_sec.h5'.format(self.directory, self.clock.times[time_point])
+        old_source = '{}/source.{}.h5'.format(self.directory, self.settings_file.batches)
+        new_source = '{}/source.h5'.format(self.directory)
+
+        # Run OpenMC
+        if not self.use_pregenerated_sps:
+            if self.run_on_cluster:
+
+                # Copy job file to run directory
+                cmd_str = 'cp ' + self.job_file  + ' ' + self.directory
+                subprocess.Popen(cmd_str, shell=True)
+                time.sleep(5)
+
+                 # Launch job
+                cmd_str = 'qsub -P moose ' + self.job_file
+                proc = subprocess.Popen(cmd_str, cwd=self.directory, stdout=subprocess.PIPE, shell=True)
+
+                 # Get the job number
+                job_name = proc.stdout.readlines()[0]
+                job_number = job_name.split('.')[0]
+
+                 # Pause and wait for run to finish
+                cmd_str = 'qstat | grep ' + str(job_number)
```

Comment:
likewise, `slurm` would use `squeue` here. And I'm sure there are lots of other schedulers with different commands.